master/internal/api_trials.go

```diff
@@ -1385,9 +1385,22 @@ func (a *apiServer) ReportCheckpoint(
 	default:
 	}
 
-	if err := db.AddCheckpointMetadata(ctx, c); err != nil {
+	task, err := db.TaskByID(ctx, model.TaskID(req.Checkpoint.TaskId))
+	if err != nil {
+		return nil, fmt.Errorf("looking up task to decide if trial: %w", err)
+	}
+	if task.TaskType != model.TaskTypeTrial {
```

Comment:
i think we have, but i would triple check there isn't a customer doing batch inference with commands and checkpointing.

Reply:
We broke this use case in https://github.com/determined-ai/determined/blob/337174635035a5edeab956ae23cd979274f271ec/master/static/migrations/20231011104015_add-exp-trials-on-del-casc.tx.up.sql#L47

It isn't too much work to fix it here I wrote a test for it here but if we broke support for it I don't see why this change should add back in support for it.
https://github.com/determined-ai/determined/pull/8535/files#diff-3ae51e125f14c5b44fa5541d9ecb8f57332daf504ec62068d141e82e8c742fe8R590